project_file_agents.py

```
@@ -0,0 +1,36 @@
+
+@ell.complex(client=openai_client, model="gpt-4o-mini", temperature=0.0, tools=[rewrite_test_project_file])
+def add_project_references(test_project_file_path: str, project_paths_to_add_to_references: list[str]):
+    """
+    Your only responsibility is to add project references to the csproj test project file that are given to you.
+    You are not allowed to add any other project references or remove any project references.
+    Use the rewrite_project_file tool to rewrite the test project file.
+    """
+import os
+from typing import NoReturn
+
+def add_project_references(test_project_file_path: str, project_paths_to_add_to_references: list[str]):
```

Ruff

Redefinition of unused `add_project_references` from line 6

(F811)